File: server/src/models/recipes/RecipeNutritionModel.js
class RecipeNutritionModel {
  constructor({ calories, carbohydrate, fat, protein }) {
    this.calories = calories;
    this.carbohydrate = carbohydrate;
    this.fat = fat;
    this.protein = protein;
  }

  static fromJson(json) {
    return new RecipeNutritionModel({
      calories: json["calories"] ? parseInt(json["calories"]) : null,
      carbohydrate: json["carbohydrate"] ? parseFloat(json["carbohydrate"]) : null,
      fat: json["fat"] ? parseFloat(json["fat"]) : null,
      protein: json["protein"] ? parseFloat(json["protein"]) : null,
    });
  }
}

export default RecipeNutritionModel;

File: lib/data/models/recipe/recipe_nutrition_model.dart
class RecipeNutritionModel {
  const RecipeNutritionModel({
    this.calories,
    this.carbs,
    this.fat,
    this.protein,
  });

  factory RecipeNutritionModel.fromJson(Map<String, dynamic> json) {
    return RecipeNutritionModel(
      calories: json['calories'] as int?,
      carbs: json['carbs'] as double?,
      fat: json['fat'] as double?,
      protein: json['protein'] as double?,
    );
  }

  final int? calories;
  final double? carbs;
  final double? fat;
  final double? protein;
}
